lib/libzrepl/data_conn.c

```diff
@@ -670,10 +677,31 @@ uzfs_zvol_rebuild_dw_replica(void *arg)
 			 * one of them might have changed rebuild state
 			 */
 			if (uzfs_zvol_get_rebuild_status(zinfo->main_zv) !=
-			    ZVOL_REBUILDING_AFS)
+			    ZVOL_REBUILDING_AFS) {
 				uzfs_zvol_set_rebuild_status(zinfo->main_zv,
 				    ZVOL_REBUILDING_AFS);
+				/*
+				 * Lets ask io_receiver thread to flush
+				 * all outstanding IOs in taskq
+				 */
+				zinfo->quiesce_done = 0;
+				zinfo->quiesce_requested = 1;
+			}
 			mutex_exit(&zinfo->main_zv->rebuild_mtx);
+			/*
+			 * Wait for all outstanding IOs to be flushed
+			 * to disk before making further progress
+			 */
+#ifdef DEBUG
+			if (inject_error.delay.rebuid_io_quiesce_check_by_pass)
+				continue;
+#endif
+			while (1) {
+				if (!zinfo->quiesce_done)
+					sleep(1);
```

Comment:
I have a doubt. If replica gets disconnected from the target during rebuilding and quiescing was requested then will this thread exit?